third_party/move/move-vm/runtime/src/storage/loader/native.rs

```diff
@@ -63,6 +64,24 @@ impl<'a> StructDefinitionLoader for NativeLoader<'a> {
     }
 }
 
+impl<'a> ModuleMetadataLoader for NativeLoader<'a> {
+    fn load_module_metadata(
+        &self,
+        _gas_meter: &mut impl GasMeter,
+        traversal_context: &mut impl ModuleTraversalContext,
+        module_id: &ModuleId,
+    ) -> PartialVMResult<Vec<Metadata>> {
+        // If used for lazy loading in native context, ensure module has been charged gas.
```

Comment:
this is a check instead of a charge so somewhere we guarantee that it's already charged?